src/ecpool_worker.erl

```diff
@@ -98,6 +105,14 @@ exec_async(Pid, Action, Callback) ->
 is_connected(Pid) ->
     gen_server:call(Pid, is_connected, infinity).
 
+-spec take_start_result(pid()) -> start_result().
```

Comment:
why take but not get ?

Reply:
Just because the `result` neither represents the current `connection state`, nor the `result of last connection attempt`. It only represents the return value of the `first connection attempt`, i.e. the result of starting the worker.

Additionally, I don't want to keep it in the proc dict to avoid potential misunderstanding during troubleshooting. For instance, seeing `{start_result, started}` might lead to the incorrect assumption that the worker is currently connected, event through it might actually be disconnected.